tripal_chado/src/Plugin/ChadoBuddy/ChadoCvtermBuddy.php

```diff
@@ -0,0 +1,1016 @@
+
+/**
+ * Plugin implementation of the chado_buddy.
+ *
+ * @ChadoBuddy(
+ *   id = "chado_cvterm_buddy",
+ *   label = @Translation("Chado Controlled Vocabulary Term Buddy"),
+ *   description = @Translation("Provides helper methods for managing chado cvs and cvterms.")
+ * )
+ */
+class ChadoCvtermBuddy extends ChadoBuddyPluginBase implements ChadoBuddyInterface, ContainerFactoryPluginInterface {
+
+  /**
+   * Used to store the manager so we can create a buddy
+   */
+  protected object $buddy_manager;
+
+  /**
+   * Provide the dbxref instance
+   */
+  protected object $dbxref_instance;
+
+ /**
+   * Implements ContainerFactoryPluginInterface->create().
+   *
+   * We are injecting an additional dependency here, the
+   * ChadoBuddyPluginManager, so that this buddy can have
+   * access to the Dbxref buddy.
+   *
+   * Since we have implemented the ContainerFactoryPluginInterface this static function
+   * will be called behind the scenes when a Plugin Manager uses createInstance(). Specifically
+   * this method is used to determine the parameters to pass to the contructor.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   * @param array $configuration
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('tripal_chado.database'),
+      $container->get('tripal_chado.chado_buddy')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition,
+                              ChadoConnection $connection, ChadoBuddyPluginManager $buddy_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $connection);
+    $this->buddy_manager = $buddy_manager;
+  }
+
+  /**
+   * Retrieves a controlled vocabulary.
+   *
+   * @param array $conditions
+   *   An array where the key is a column in chado and the value describes the
+   *   cv you want to select. Valid keys include:
+   *     - cv.cv_id
+   *     - cv.name
+   *     - cv.definition
+   *     - buddy_record = a ChadoBuddyRecord can be used
+   *       in place of or in addition to other keys
+   * @param array $options (Optional)
+   *   Associative array of options.
+   *     - 'case_insensitive' - a single key, or an array of keys
+   *                            to query case insensitively.
+   *
+   * @return bool|array|ChadoBuddyRecord
+   *   If the select values return a single record then we return the
+   *     ChadoBuddyRecord describing the chado record.
+   *   If the select values return multiple records, then we return an array
+   *     of ChadoBuddyRecords describing the results.
+   *   If there are no results then we return FALSE.
```

Comment:
After discussion we think we're going to change this. It was written this way based on the old API functions but we've decided to change the return value to simplify code in the future even if it makes it a bit harder to upgrade to these functions.
```suggestion
   * @return array
   *   An array of ChadoBuddyRecord objects. More specifically,
   *   (1) if the select values return a single record then we return an 
   *     array containing a single ChadoBuddyRecord describing the record.
   *   (2) if the select values return multiple records, then we return an 
   *     array of ChadoBuddyRecords describing the results.
   *   (3) if there are no results then we return an empty array.
```

Reply:
Marked this unresolved since the code change to the docblock was committed but the change to the code itself has not yet been made and we don't want to lose this.

Reply:
Implemented in code, and so this can be resolved